### PR TITLE
Add notable release highlights for 7.0

### PR DESCRIPTION
**Pull request by jasontedor**
This commit adds notable release highlights for 7.0.

### CODE REVIEW

**Comment by jasontedor** [comment]
@jpountz @eskibars There is no content for us to reuse here (i.e., there's more work to do here than merely going through existing release highlights and tagging the ones that we think are notable). Therefore, as a first step I have created a list of what I think are the notable release highlights for 7.0.0. Let us use this to iterate together on what we think the list should be? Once we have agreement on the list, we can craft a plan for producing the content that would back these items. Much like #40478 I have tried to keep this list from getting too long.

File: docs/reference/release-notes/highlights-7.0.0.asciidoc [hunk 1]
```diff
@@ -13,5 +13,97 @@ See also <<breaking-changes-7.0,Breaking changes>> and
...
+// end::notable-highlights[]
+
+// tag::notable-highlights[]
+=== Support nanosecond timestamps
+
+Words.
+// end::notable-highlights[]
+
+// tag::notable-highlights[]
+=== Disabling hit counts by default
```
Comment (jpountz) [inline docs/reference/release-notes/highlights-7.0.0.asciidoc]: this is the breaking part of the change, I think a better description would rather be something like "Faster retrieval of top hits"

**Review by jpountz — COMMENTED** [review]
I'd add "rank_feature and rank_features fields". Otherwise this looks good to me.

File: docs/reference/release-notes/highlights-7.0.0.asciidoc (hunk 1 above)
Comment (jasontedor) [reply docs/reference/release-notes/highlights-7.0.0.asciidoc]: I pushed b5d322c7912bf14f50cc2a9fc1b25653a9ed5280

**Review by eskibars — COMMENTED** [review]
This is a great recap.  Two comments:
1) I think the script score query is pretty neat and I think others will as well.
2) Not sure if we want to include the GA features ILM, SQL, and CCR.  I know we've often done this before when we've dual-targeted GA versions like this, but we sometimes get some confusion around it ("I thought it was already GA").  I don't have a strong preference one way or the other though

**Comment by jasontedor** [comment]
Ha! I debated internally with myself on script score query, so I’m happy to include it.

I do think we should include these GA features that also went GA in 6.7.0. Let’s jot assume everyone is on top of our release and GA schedule is where I come from.

File: docs/reference/release-notes/highlights-7.0.0.asciidoc [hunk 2]
```diff
@@ -12,6 +12,110 @@ See also <<breaking-changes-7.0,Breaking changes>> and
...
+//end::notable-highlights[]
+
+//tag::notable-highlights[]
+=== {ilm} is production-ready
+
+Words.
+//end::notable-highlights[]
+
+//tag::notable-highlights[]
+=== SQL is production-ready
```
Comment (colings86) [inline docs/reference/release-notes/highlights-7.0.0.asciidoc]: SQL was made GA in 6.7.0, do we need to mention it for 7.0 too?

File: docs/reference/release-notes/highlights-7.0.0.asciidoc [hunk 3]
```diff
@@ -12,6 +12,110 @@ See also <<breaking-changes-7.0,Breaking changes>> and
...
+//end::notable-highlights[]
+
+//tag::notable-highlights[]
+=== Support nanosecond timestamps
+
+Words.
+//end::notable-highlights[]
+
+//tag::notable-highlights[]
+=== Faster retrieval of top hits
```
Comment (colings86) [inline docs/reference/release-notes/highlights-7.0.0.asciidoc]: Is this where we will mention total hits not being tracked by default or should we make a new heading for that?

**Review by colings86 — COMMENTED** [review]
I left a couple of comments. Thanks for doing this @jasontedor

File: docs/reference/release-notes/highlights-7.0.0.asciidoc (hunk 2 above)
Comment (colings86) [reply docs/reference/release-notes/highlights-7.0.0.asciidoc]: This was answered in another comment here: https://github.com/elastic/elasticsearch/pull/40481#issuecomment-476916583

**Review by jpountz — APPROVED** [review]

**Comment by lcawl** [comment]
Forward-fit to 7.x via edb0d42b41f21838a0d142dac4f64b91e4055b88

**Pull request merged**